### PR TITLE
Fix the dataMoverTopic config param always being required

### PR DESCRIPTION
**Pull request by welansari**
Extension configs are validated independently of each other, so validating the dataMoverTopic based on the supportedLifecycleRules param is not currently possible.

Issue: BB-619

### CODE REVIEW

File: extensions/replication/ReplicationConfigValidator.js [hunk 1]
```diff
@@ -53,11 +53,7 @@ const joiSchema = joi.object({
         bootstrapList: bootstrapListJoi,
     },
     topic: joi.string().required(),
-    dataMoverTopic: joi.string().when('..lifecycle.supportedLifecycleRules', {
-        is: joi.array().has(joi.string().valid('transition')),
-        then: joi.required(),
-        otherwise: joi.forbidden(),
-    }),
+    dataMoverTopic: joi.string().optional(),
```
Comment (francoisferrand) [inline extensions/replication/ReplicationConfigValidator.js:56]: can we move this validation somewhere else anyway?

**Review by KillianG — APPROVED** [review]

File: tests/unit/lib/config/Config.spec.js [hunk 2]
```diff
@@ -6,20 +6,47 @@ const { Config } = require('../../../../lib/Config');
 const backbeatConfig = require('./config.json');
 
 describe('Config', () => {
+    let config;
+    let testConfig;
+
+    beforeEach(() => {
+        config = new Config();
+        // deep copy the config to avoid modifying the original
+        testConfig = JSON.parse(JSON.stringify(backbeatConfig));
```
Comment (ghost) [inline tests/unit/lib/config/Config.spec.js:15]: Nit
```suggestion
        testConfig = { ...backbeatConfig };
```

**Review by ghost — APPROVED** [review]

File: tests/unit/lib/config/Config.spec.js (hunk 2 above)
Comment (welansari) [reply tests/unit/lib/config/Config.spec.js:15]: `{ ...backbeatConfig }` doesn't deep copy nested objets.

**Pull request merged**